### DISCUSSION TITLE
Workflow configurations to automatically publish the app after it has been approved

### DISCUSSION THREAD

**Discussion by MbuthiaWaKihara** [codemagic-ci-cd/codemagic-docs#2873] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Probably a simple thing I'm missing, but I'm having a hard time getting it from the docs.

How can I configure my Android and iOS workflow, so that if the app is approved on Google Play and App Store, it is automatically published, and I don't have to manually publish the app?

I think for iOS, under `publishing`, and `app_store_connect`, I can add `release_type` and set it to `AFTER_APPROVAL`, but I'm not sure. Please help.

### Expected results

App is automatically published after it has been approved.

### Actual results

I have to publish app after it has been approved

**Comment by dam-ease** [comment, marked as answer]
Hello @MbuthiaWaKihara,
Yes, that's correct, `release_type` parameter under `app_store_connect` in the publish section helps with this. For iOS publishing, you can check out more details [here](https://docs.codemagic.io/yaml-publishing/app-store-connect/#setting-up-publishing-to-app-store-connect). Setting `release_type: AFTER_APPROVAL` means the app is automatically released once Apple approves it. Also, make sure the `submit_to_app_store` flag is set to true to send the app for review.
`release_promotion` works in the case of Google Play publishing. You can check out more details [here](https://docs.codemagic.io/yaml-publishing/google-play/#configure-publishing-in-codemagicyaml).

**Reply by MbuthiaWaKihara** [reply]
Hello @dam-ease 

`release_type: AFTER_APPROVAL` worked fine for iOS. For Android however, under `google_play`, I added `release_promotion: ` and under it I added `track: production`, but I still had to manually publish after approval. Is there something I am missing?

```
      google_play:
        credentials: $GCLOUD_SERVICE_ACCOUNT_CREDENTIALS
        track: production
        submit_as_draft: false
        release_promotion:
          track: production
```

**Reply by dam-ease** [reply]
Hello @MbuthiaWaKihara,
What's left to do here is in your Google Play account. Use the Publishing overview page to turn managed publishing on or off. When managed publishing is on, you control when approved changes are published on Google Play. If managed publishing is off, changes are published as soon as they're approved.
<img width="1231" alt="Screenshot 2024-12-13 at 15 39 52" src="https://github.com/user-attachments/assets/92c3e735-76b4-4a6b-ab14-f09bdd845747" />
